### ISSUE TITLE
storage: ReplicaGC took longer than a minute during decommissioning on indigo

### ISSUE THREAD

**Issue by a-robinson** [cockroachdb/cockroach#17848]
After doing a lot of splitting on indigo (50k+ ranges), I decided to try decommissioning and re-adding a couple nodes to test out how rebalancing performs when adding multiple new nodes to a large cluster. In the process of decommissioning the first node, I've seen a few replicaGC attempts time out looking up the range descriptor:

```
W170823 15:27:31.333934 437849953 kv/dist_sender.go:1156  [replicaGC,s9,r16441/3:/Table/51/1/46879{2985…-4436…},n9,range-lookup] have been waiting 1m0s sending RPC to r1 for batch: RangeLookup [/Meta1/Tab
le/51/1/4687929853182520574,/Min)
E170823 15:27:31.334948 264 storage/queue.go:656  [replicaGC,n9,s9,r16441/3:/Table/51/1/46879{2985…-4436…}] context deadline exceeded
I170823 15:27:31.334973 264 storage/queue.go:536  [replicaGC,n9,s9,r16441/3:/Table/51/1/46879{2985…-4436…}] done 1m0.012170327s
```

```
I170823 15:27:31.528867 264 storage/replica.go:736  [replicaGC,n9,s9,r16441/3:/Table/51/1/46879{2985…-4436…}] removed 660 (649+11) keys in 12ms [clear=0ms commit=11ms]
I170823 15:27:31.528920 264 storage/queue.go:635  [replicaGC,n9,s9,r16441/3:/Table/51/1/46879{2985…-4436…}] done
I170823 15:27:31.528946 264 storage/queue.go:536  [replicaGC,n9,s9,r16441/3:/Table/51/1/46879{2985…-4436…}] done 130.349573ms
```

It happened to another replica as well, although without any log messages about a slow RangeLookup request:

```
E170823 15:30:27.616377 264 storage/queue.go:656  [replicaGC,n9,s9,r2225/1:/System/tsd/cr.store.raft.pro…] context deadline exceeded
I170823 15:30:27.616464 264 storage/queue.go:536  [replicaGC,n9,s9,r2225/1:/System/tsd/cr.store.raft.pro…] done 1m0.000720947s
```

```
I170823 15:30:27.778204 264 storage/store.go:2146  [replicaGC,n9,s9,r2225/1:/System/tsd/cr.store.raft.pro…] removing replica
I170823 15:30:27.779777 264 storage/replica.go:736  [replicaGC,n9,s9,r2225/1:/System/tsd/cr.store.raft.pro…] removed 1934 (1922+12) keys in 1ms [clear=0ms commit=1ms]
I170823 15:30:27.779829 264 storage/queue.go:635  [replicaGC,n9,s9,r2225/1:/System/tsd/cr.store.raft.pro…] done
I170823 15:30:27.779856 264 storage/queue.go:536  [replicaGC,n9,s9,r2225/1:/System/tsd/cr.store.raft.pro…] done 105.074821ms
```

I don't think this is the same as #16266 given what I know about that, but it may be connected given that this is happening after a bunch of splits. It's worth noting that the cluster's performance was already degrading after the splits but prior to me starting the decommissioning process, so something was likely already wrong, but nothing appears to be stuck. The cluster is running  050bc89f248ee761af4852798a50fa2514646c59.

@tschottdorf

**Comment by bdarnell** [comment]
Sounds like #17760, fixed in #17766

**Comment by a-robinson** [comment]
Yeah, that seems likely. I've bumped the build indigo is running and will reopen if it reoccurs.
